src/ameba/ast/visitors/scope_visitor.cr

```diff
@@ -3,6 +3,21 @@ require "./base_visitor"
 module Ameba::AST
   # AST Visitor that traverses the source and constructs scopes.
   class ScopeVisitor < BaseVisitor
+    SUPER = "super"
+    # List of special top level macros that use assignments
+    # as the macro arguments thus such arguments are not treated as assignments.
```

Comment:
I'd add a note that this is essentially just guessing that such a call (like "record") probably calls a macro.